lib/src/select/Select.tsx

```diff
@@ -360,6 +281,12 @@ const DxcSelect = React.forwardRef<RefType, SelectPropsType>(
       setSearchValue("");
     };
 
+    const handleOptionOnClick = useCallback(() => {
+      handleSelectChangeValue(optionalEmptyOption);
+      !multiple && closeOptions();
+      setSearchValue("");
+    }, [handleSelectChangeValue, closeOptions, multiple]);
+
```

Comment:
What if we make this function generic to avoid repetition of code?
We can pass the option as an argument like:

```
const handleOptionOnClick = useCallback((option) => {
      handleSelectChangeValue(option);
      !multiple && closeOptions();
      setSearchValue("");
    }, [handleSelectChangeValue, closeOptions, multiple]);
```

Reply:
Yes! I started with the strategy you mention in the comments (a common function surrounded by a `useCallback` to not cause unnecessary re-renders) but, for some reason, i discontinued it.